module-info are added; structrure is updated to conform M2

solomax opened the pull request.

solomax committed:
module-info are added; structrure is updated to conform M2

solomax committed:
Code clean-up

martin-g commented on bootstrap-sass/pom.xml:
```diff
@@ -14,14 +14,12 @@
     <dependencies>
         <dependency>
-            <groupId>ua.kasta</groupId>
+            <groupId>io.bit3</groupId>
```
I think we moved from `bit3` to `kasta` because `bit3` was abandoned and it does not support Mac ARM64.
See https://github.com/l0rdn1kk0n/wicket-bootstrap/pull/986

reckart replied:
Mainly because of ARM - I believe it continues being abandoned in any case. The native library that is being wrapped here is no longer developed as far as I remember...

solomax replied:
I'll revert this change

BUT according to https://mvnrepository.com/artifact/ua.kasta/jsass `ua.kasta:jsass` seems to be dead as well :(((
It is "one-release-repo" :((((

reckart replied:
There is little point in maintaining a java wrapper for a library which is dead. See notice on jsass repo:

> libsass has been [declared deprecated](https://sass-lang.com/blog/libsass-is-deprecated) and there will be no further development! This also ends the further development of jsass. It is not yet clear whether there will ever be an implementation based on the [Sass embedded protocol](https://github.com/sass/embedded-protocol). I recommend switching to [Webpack](https://webpack.js.org/) or similar technologies! This project will continue to be maintained for the time being and compatibility with newer Java versions will be ensured. But please do not expect any further development on my part.

reckart replied:
fwiw I have switched to using sass with the maven-frontend-plugin some time back.

solomax replied:
> I have switched to using sass with the maven-frontend-plugin

I did the same :)

solomax committed:
`io.bit3` -> `ua.kasta`

martin-g commented:
@solomax Anything else you want to add here or it is ready to be merged ?

solomax commented:
> @solomax Anything else you want to add here or it is ready to be merged ?

Let's wait for the M2 release to drop `SNAPSHOT` dependency :)
Meanwhile I'll try to create PRs for `jquery.selectors` and `wicket.webjars` so they will have Automatic module anmes (or be converted to JMS)

solomax committed:
Dependencies are updated

martin-g committed:
Get rid of the dependency to WicketStuff-annotation

This will make it easier to release.
Currently WB depends on WS-Annotation, and some other WS projects depend
on WB.

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@solomax Ready to go ?

martin-g commented:
I wonder whether we need to add `open`s too ? Like WICKET-7081

solomax commented:
> I wonder whether we need to add `open`s too ? Like WICKET-7081

I'll try to check later this week or early next week :)

solomax commented:
> @solomax Ready to go ?

yes IMO this one can be released (so people can test it :)))

solomax commented on bootstrap-samples/src/main/java/de/agilecoders/wicket/samples/WicketApplication.java:
```diff
@@ -93,7 +111,7 @@ public void init() {
 
         optimizeForWebPerformance();
 
-        new AnnotatedMountScanner().scanPackage("de.agilecoders.wicket.samples.pages").mount(this);
```
Actually this class wasn't dropped: https://github.com/wicketstuff/core/blob/master/annotation/src/main/java/org/wicketstuff/annotation/scan/AnnotatedMountScanner.java

my local build is green ... :(

I can run `bootstrap-samples` using `mvn jetty:run`

```
Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)
Maven home: /opt/maven
Java version: 17.0.8.1, vendor: Private Build, runtime: /usr/lib/jvm/java-17-openjdk-amd64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.15.0-86-generic", arch: "amd64", family: "unix"
```

martin-g commented:
On Thu, 19 Oct 2023 at 19:11, Maxim Solodovnik ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In
> bootstrap-samples/src/main/java/de/agilecoders/wicket/samples/WicketApplication.java
> <https://github.com/l0rdn1kk0n/wicket-bootstrap/pull/1004#discussion_r1365803784>
> :
>
> > @@ -93,7 +111,7 @@ public void init() {
>
>          optimizeForWebPerformance();
>
> -        new AnnotatedMountScanner().scanPackage("de.agilecoders.wicket.samples.pages").mount(this);
>
> Actually this class wasn't dropped:
> https://github.com/wicketstuff/core/blob/master/annotation/src/main/java/org/wicketstuff/annotation/scan/AnnotatedMountScanner.java
>
> my local build is green ... :(
>

Why green build is a bad thing ? :-)

I just removed the usage of AnnotatedMountScanner in WB.
Now there is no dependency to wicketstuff.

I can run bootstrap-samples using mvn jetty:run
>
> Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)
> Maven home: /opt/maven
> Java version: 17.0.8.1, vendor: Private Build, runtime: /usr/lib/jvm/java-17-openjdk-amd64
> Default locale: en_US, platform encoding: UTF-8
> OS name: "linux", version: "5.15.0-86-generic", arch: "amd64", family: "unix"
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/l0rdn1kk0n/wicket-bootstrap/pull/1004#pullrequestreview-1688152450>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABYUQTUJRNP6RXXAGYVHS3YAFGMTAVCNFSM6AAAAAA6AYZRQGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMOBYGE2TENBVGA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

solomax commented:
> Why green build is a bad thing ? :-) I just removed the usage of AnnotatedMountScanner in WB. Now there is no dependency to wicketstuff. I can run bootstrap-samples using mvn jetty:run

green build is not so bad :)
different results - is something need to be investigated IMO :))

martin-g commented:
Ah, you meant the issue with `maven-source-plugin`.
`mvn help:effective-pom` showed that there are two executions for this plugin. One of them comes from the parent pom (Sonatype OSS) - with id `attach-sources`

martin-g commented:
> yes IMO this one can be released

Done!
